corehq/apps/reports/v2/endpoints/case_properties.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import json
+from memoized import memoized
+
+from corehq.apps.case_search.const import (
+    SPECIAL_CASE_PROPERTIES,
+    CASE_COMPUTED_METADATA,
+)
+from corehq.apps.reports.standard.cases.filters import (
+    get_flattened_case_properties,
+)
+from corehq.apps.reports.v2.models import BaseOptionsEndpoint
+
+
+class CasePropertiesEndpoint(BaseOptionsEndpoint):
+    slug = "case_properties"
+
+    @property
+    @memoized
+    def report_context(self):
+        return json.loads(self.data.get('reportContext', "{}"))
+
+    @property
+    @memoized
+    def existing_slugs(self):
+        return self.report_context.get('existingSlugs', [])
+
+    @property
+    def excluded_slugs(self):
+        return ['name'] + self.existing_slugs
+
+    @property
+    @memoized
+    def case_properties(self):
+        case_properties = get_flattened_case_properties(
+            self.domain, include_parent_properties=False
+        )
+        special_properties = [
+            {'name': prop, 'case_type': None, 'meta_type': 'info'}
+            for prop in SPECIAL_CASE_PROPERTIES + CASE_COMPUTED_METADATA
+        ]
+        return case_properties + special_properties
+
+    @property
+    def filtered_case_properties(self):
+
+        def _filter_property(prop):
```

Comment:
if a specific `case_type` were selected in the global filters (coming soon!) I would want to make sure that properties from other case types don't show up here...

Reply:
I'm asking because currently this is handled on the front end so that the report doesn't have to make round trips to the server just to filter a list.

Reply:
I'd imagine doing more complex filtering in the future when fetching column options. Also, it's not doing this round trip often. I wouldn't say it's expensive, and it allows for greater flexibility moving forward.